### PR TITLE
[iOS] Label HorizontalTextAlignment="Center" not working in conjunction with LineHeight on iOS

### LINKED ISSUE #4262: Label HorizontalTextAlignment="Center" not working in conjunction with LineHeight on iOS
### Description

HorizontalTextAlignment="Center" has no effect if LineHeight is assigned a value

### Steps to Reproduce

1. Create 2 labels 
2.
``` <Label HorizontalTextAlignment="Center" Text="This is center aligned&#x0a;line 2" />```
```<Label HorizontalTextAlignment="Center" LineHeight="1.5"  Text="This should be center aligned&#x0a;line 2"  />```
 

### Expected Behavior
Both labels center aligned
### Actual Behavior
On Android both labels are center aligned
On iOS only the first label is center aligned
### Basic Information

- Version with issue: 3.3.0.912540
- Last known good version: N/A issue found when migrating from custom renderer
- IDE: VS on Mac
- Platform Target Frameworks: 
  - iOS:   11.1
  - Android:  7.1
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by masonyc**
Label HorizontalTextAlignment="Center" not working in conjunction with LineHeight on iOS

### Description of Change ###

Changed span.ToAttributed method to set the text alignment for the attributed text in iOS.

### Issues Resolved ### 
- fixes #4262 

### API Changes ###
None

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
In iOS, when Horizontal Text Alignment and Line Height both set now will do the alignment correctly on label.

### Before/After Screenshots ### 
Before
previous

After 
after

### Testing Procedure ###
Start ControlGallery.iOS, Search and go to G4262 Page in tests screen, Two labels should both be centered.

### CODE REVIEW

**Comment by samhouts** [comment]
build --uitests

File: Xamarin.Forms.Platform.iOS/Renderers/FormattedStringExtensions.cs [hunk 1]
```diff
@@ -63,14 +63,47 @@ internal static NSAttributedString ToAttributed(this Span span, Element owner, C
...
-			NSMutableParagraphStyle style = null;
+			NSMutableParagraphStyle style = new NSMutableParagraphStyle();
 			lineHeight = span.LineHeight >= 0 ? span.LineHeight : lineHeight;
 			if (lineHeight >= 0)
 			{
-				style = new NSMutableParagraphStyle();
 				style.LineHeightMultiple = new nfloat(lineHeight);
 			}
 
+#if __MOBILE__
```
Comment (samhouts) [inline Xamarin.Forms.Platform.iOS/Renderers/FormattedStringExtensions.cs]: We can simplify this by adding `using UITextAlignment = AppKit.NSTextAlignment;` inside the `#else` definition at the top of the file, I think. Then we only need this code written once without the `#if` here.

Comment (masonyc) [reply Xamarin.Forms.Platform.iOS/Renderers/FormattedStringExtensions.cs]: @samhouts Hi, I have done the changes as requested. Can you please have another check, thanks.

Comment (samhouts) [reply Xamarin.Forms.Platform.iOS/Renderers/FormattedStringExtensions.cs]: That was fast! Thanks!

**Comment by samhouts** [comment]
build --uitests

**Comment by masonyc** [comment]
The build failed, what was the error?

**Comment by samhouts** [comment]
hmm, it was internal. let me try again!

**Comment by samhouts** [comment]
build --uitests

**Comment by hartez** [comment]
> The build failed, what was the error?

Looks like it's not building because you remove the Xamarin.Forms.Build submodule. Revert those changes on your PR and it should build.

**Comment by masonyc** [comment]
@hartez @samhouts Sorry, I have reverted the changes now.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue4262.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 4262, "Label HorizontalTextAlignment=\"Center\" not working in conjunction with LineHeight on iOS", PlatformAffected.iOS)]
+	public class Issue4262 : ContentPage
+	{
+		public Issue4262()
+		{
+			var label = new Label() { Text = "This is center aligned&#x0a;line 2.", HorizontalTextAlignment = TextAlignment.Center };
+			var label2 = new Label() { Text = "This is center aligned&#x0a;line 2.", HorizontalTextAlignment = TextAlignment.Center, LineHeight = 1.5 };
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue4262.cs]:
```suggestion
			var label2 = new Label() { Text = "If this is not center aligned, this test has failed.", HorizontalTextAlignment = TextAlignment.Center, LineHeight = 1.5 };
```

Comment (hartez) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue4262.cs]: Make the test case more explicit for anyone who has to manually run it (since we can't easily automate this test).

Comment (masonyc) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue4262.cs]: done
